File: Go-Solutions/226.go
package main

import "Go-Solutions/ds"

func invertTree(root *ds.TreeNode) *ds.TreeNode {
	if root == nil {
		return nil
	}
	invertTree(root.Left)
	invertTree(root.Right)
	root.Left, root.Right = root.Right, root.Left
	return root
}

File: Java-Kotlin/src/main/java/tree/_226/_226.java
package tree._226;

import common.datastructure.TreeNode;

/**
 * Problem link: https://leetcode-cn.com/problems/invert-binary-tree/submissions/
 * <p>
 * Solution: just post order traversal
 */
public class _226 {

    public TreeNode invertTree(TreeNode root) {
        helper(root);
        return root;
    }

    private void helper(TreeNode root) {
        if (root == null) {
            return;
        }
        helper(root.left);
        helper(root.right);
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;
    }
}
